pkg/ccl/crosscluster/physical/stream_ingestion_planning.go

```diff
@@ -291,6 +300,47 @@ func createReplicationJob(
 	return err
 }
 
+func createReaderTenant(
+	ctx context.Context,
+	p sql.PlanHookState,
+	tenantInfo mtinfopb.TenantInfoWithUsage,
+	destinationTenantID roachpb.TenantID,
+	options *resolvedTenantReplicationOptions,
+) (roachpb.TenantID, error) {
+	var readerID roachpb.TenantID
+	if options.ReaderTenantEnabled() {
+		var readerInfo mtinfopb.TenantInfoWithUsage
+		readerInfo.DataState = mtinfopb.DataStateAdd
+		readerInfo.Name = tenantInfo.Name + "-readonly"
+		readerInfo.ReadFromTenant = &destinationTenantID
+
+		readerZcfg, err := sql.GetHydratedZoneConfigForTenantsRange(ctx, p.Txn(), p.ExtendedEvalContext().Descs)
+		if err != nil {
+			return readerID, err
+		}
+
+		readerID, err = sql.CreateTenantRecord(
+			ctx, p.ExecCfg().Codec, p.ExecCfg().Settings,
+			p.InternalSQLTxn(),
+			p.ExecCfg().SpanConfigKVAccessor.WithISQLTxn(ctx, p.InternalSQLTxn()),
+			&readerInfo, readerZcfg,
+			false, p.ExecCfg().TenantTestingKnobs,
+		)
+		if err != nil {
+			return readerID, err
+		}
+
+		readerInfo.ID = readerID.ToUint64()
+		if err := p.ExecCfg().DB.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
```

Comment:
oh, didn't catch this the first time. I think we need to pass `p.Txn()` to BootstrapTenant below, instead of creating a new txn. The planHook already has a txn associated with it that we're supposed to commit all state with. Using a new txn could cause deadlock.